src/main/java/teammates/storage/sqlapi/UsersDb.java

```diff
@@ -277,4 +280,46 @@ public List<Student> getAllStudentsForEmail(String email) {
         return HibernateUtil.createQuery(cr).getResultList();
     }
 
+    /**
+     * Resets instructor's account by {@code userEmail} and {@code courseId}.
+     */
+    public void resetInstructorGoogleId(String userEmail, String courseId)
+            throws EntityDoesNotExistException {
+        assert userEmail != null;
+        assert courseId != null;
+
+        Instructor instructor = getInstructorForEmail(courseId, userEmail);
+
+        if (instructor == null) {
+            // What is a better message?
+            throw new EntityDoesNotExistException(ERROR_UPDATE_NON_EXISTENT +
+                    "Instructor [courseId=" + courseId + ", email=" + userEmail + "]");
+        }
+
+        instructor.setAccount(null);
```

Comment:
Logic should be in logic layer. Simply get instructor, instructor.setAccount(null) then cascade the update if necessary. No need to call merge anymore.

Reply:
Samuel, I did up some reading. As said before, Account is the parent of User (be it Instructor/Student).

User and Account only have a unidirectional relationship, ie. User has an attribute accountId which tells us which Account this User to referring to.

I was thinking of using any of the `Cascade` annotation maybe `C.ALL` or `C.MERGE` (not exactly sure..). All the examples I have seen so far uses Cascade from parent to child and some of which are bidirectional, but we are doing child to parent.

Not exactly sure how to cascade the update or what do you mean by `if necessary`